apps/src/templates/instructions/utils.js

```diff
@@ -35,6 +35,40 @@ export function scrollTo(element, scrollTop, animate=400) {
   }
 }
 
+/**
+ * Converts any inline XML in the container element into embedded
+ * readonly BlockSpaces
+ * @param {Element} container The element in which to search for XML
+ */
+export function convertXmlToBlockly(container) {
+  const xmls = container.getElementsByTagName('xml');
+  Array.prototype.forEach.call(xmls, function (xml) {
+    // create a container and insert the blockspace into it
+    const container = xml.parentNode.insertBefore(document.createElement('div'), xml);
+    let blockSpace;
+    try {
+      blockSpace = Blockly.BlockSpace.createReadOnlyBlockSpace(container, xml, {
+        noScrolling: true
+      });
+    } catch (e) {
+      // This could error out for many reaons, from invalid XML to
+      // simply misconfigured blocks. In any case, treat an error as a
+      // noop
+      return;
```

Comment:
I wonder if it's worth letting this error be thrown?  Otherwise it may be hard to tell why the block is only white space / failing silently.  When we have misconfigured blocks in other parts of the level (starting blocks, etc.) I believe we throw the error.